cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedule.java

```diff
@@ -26,14 +26,47 @@
 
   private final String description;
 
+  // NOTE: the two below attributes are left for backwards compatibility
   private final String cronEntry;
 
+  /**
+   * Build a time-based schedule.
+   *
+   * @param name name of the schedule
+   * @param description description of the schedule
+   * @param cronExpression cron expression for the schedule
+   * @return a schedule based on the given {@code cronExpression}
+   */
+  public static Schedule buildTimeSchedule(String name, String description, String cronExpression) {
```

Comment:
It was my idea with the assumption that we still keep the `Schedule` class as an abstract parent for all different kinds of schedule classes (currently would be `TimeSchedule` and `StreamSizeSchedule`). Doing it like this let the user to discover what kind of schedules are supported by the system in one place (`Schedule.` with auto completion helps) while hiding the internal classes details.